src/humanloop/decorators/tool.py

```diff
@@ -0,0 +1,189 @@
+import builtins
+import inspect
+import textwrap
+import typing
+import uuid
+from typing import Callable, Optional, Union
+from functools import wraps
+
+
+from humanloop.otel import get_trace_context, get_tracer, pop_trace_context, push_trace_context
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY, HL_TRACE_METADATA_KEY
+from humanloop.otel.helpers import write_to_opentelemetry_span
+
+from .helpers import args_to_inputs
+
+
+def _type_to_schema(type_hint):
+    if type_hint is int:
```

Comment:
What happens if a users function is not typed and or does not have good doc strings etc

Reply:
1. If docstring is not specified I use an empty string in the definition.
2. Since you've left the comment, I have extended the type parsing to take any value for unspecified type hints